doc/Language/subscripts.pod6

```diff
@@ -253,68 +251,74 @@ dimensions>) the subscript is preserved across the slice operation
-output list to contain undefined values (or L<whatever else|
+Referring to nonexistent elements in a slice subscript causes the
+output C<List> to contain undefined values (or L<whatever else|
 #Nonexistent elements> the collection in question chooses to return for
-nonexistent elements). However if the outer object passed to
-(L<one dimension of|#Multiple dimensions>) of a positional subscript is a
-L<Range>, it will be automatically truncated to the actual size of the
-collection:
-
-    my @letters = <a b c d e f>;
-    say @letters[3, 4, 5, 6, 7];  # OUTPUT: «(d e f (Any) (Any))␤»
-    say @letters[3 .. 7];         # OUTPUT: «(d e f)␤»
-
-L<From-the-end|#From the end> indices are allowed as range end-points.
+nonexistent elements):
 
-    =begin code :preamble<my @array;>
-    say @array[*-3 .. *];       # select the last three elements
+    =begin code
+    my  @letters = <a b c d e f>;
+    say @letters[3..7];  # OUTPUT: «(d e f (Any) (Any))␤»
     =end code
 
-A similar thing is done for lazy sequences, but it is often impossible to
-determine whether the sequence is infinite.  Just as often, the first part
-of the sequence is already known, and it would be silly to pretend we
-did not know it.  As a stopgap measure to prevent runaway generation of huge
-lists, a lazy subscript will not truncate as long as it does not have to
-lazily generate values, but once it starts generating values lazily, it
-will stop if it generates a value that points to a nonexistent index.
+This behavior, while at first glance may seem unintuitive, is desirable in
+instances when you want to assign a value at an index in which a defined
+value does not currently exist.
 
-    =begin code :skip-test
-    say @letters[0, 2, 4 ... *];     # Every other element of the array.
-    =end code
-
-This feature is more for protection against accidental out-of-memory
-problems than for actual use.  Since some lazy sequences cache their
-results, every time they are used in a truncation, they accumulate one
-more known element.  Things like this should probably be avoided rather
-than used for effect:
+    =begin code
+    my  @letters;
+    say @letters; # OUTPUT: «[]␤»
 
-    =begin code :skip-test
-    my @a = 2, 3 ... *;
-    say flat @letters[0, 7, @a]; # OUTPUT: «(a (Any) c d e f)␤»
-    say flat @letters[0, 7, @a]; # OUTPUT: «(a (Any) c d e f (Any))␤»
+    @letters[^10] = 'a'..'z';
+    say @letters; # OUTPUT: «[a b c d e f g h i j]␤»
     =end code
 
-The runaway protection is not perfect.  The indices are eagerly evaluated,
-with the only stop condition being truncation.  This is to provide
-mostly consistent results when there is self-reference/mutation inside
-the indices.  As such, the following will most likely hang until all
-memory has been consumed:
+If you want the resulting slice to only include defined elements, you can
+silently skip the non-defined elements using the L<#:v> adverb.
```

Comment:
*[request for change]*

Same as above, change "defined" to "existent" ("extant"?), since it doesn't care about undefinedness:

```
$ perl6 -e '(Any, Int)[1..*]:v.say'
((Int))
```